docs/wiki/blocks/block-body.md

```diff
@@ -0,0 +1,70 @@
+# Block Body
+
+General format of a **Block Body** in a Mimblewimble/Grin is basically a Mimblewimble transaction which consists of a list of **Inputs**, **Outputs** and **Kernels**.
+
+| Field         | Description           | Size                 |
+|:--------------|:----------------------|:---------------------|
+| # inputs      | Number of inputs      | 8 bytes              |
+| # outputs     | Number of outputs     | 8 bytes              |
+| # kernels     | Number of kernels     | 8 bytes              |
+| inputs        | Serialized inputs     | 34 bytes per input   |
+| outputs       | Serialized outputs    | 42 bytes per output  |
+| kernels       | Serialized kernels    | x bytes per kernel   |
+
+The kernel size **x** depends on the kernel type. More details in its dedicated section below.
+
+## Inputs
+
+Each of the inputs consists of **Feature** and a **Commitment**.
+
+| Field         | Description                             | Size                |
+|:--------------|:----------------------------------------|:--------------------|
+| Feature       | **Plain** (0x00) or **Coinbase** (0x01) | 1 byte              |
+| Commitment    | Pedersen commitment                     | 33 bytes            |
+
+## Outputs
+
+Just as in case of the inputs, the outputs consist of **Feature** and a **Commitment** but in addition they also hold **Range proofs**
+
+| Field         | Description                             | Size                |
+|:--------------|:----------------------------------------|:--------------------|
+| Feature       | **Plain** (0x00) or **Coinbase** (0x01) | 1 byte              |
+| Commitment    | Pedersen commitment                     | 33 bytes            |
+| Range proof   | Proves the transaction is net positive  | 8 bytes             |
```

Comment:
Tell me more about these 8 byte rangeproofs you invented:-!

Reply:
Right, I think it should be first 8 bytes indicate the proof size and then the proof of that indicated size, is this correct? https://github.com/GrinPlusPlus/GrinPlusPlus/blob/b7b06a1ab6c0e7f7263878f788ea7e4203dcab7c/include/Crypto/Models/RangeProof.h#L47-L54

Reply:
@tromp updated